### PR TITLE
DOC Small rewrite of interpolate recompute_scale_factor docstring

### LINKED ISSUE #55909: F.interpolate recompute_scale_factor doc is bad
## 📚 Documentation

The doc reads
> recompute_scale_factor (bool, optional) – recompute the scale_factor for use in the interpolation calculation. When scale_factor is passed as a parameter, it is used to compute the output_size. If recompute_scale_factor is False or not specified, the passed-in scale_factor will be used in the interpolation computation. Otherwise, a new scale_factor will be computed based on the output and input sizes for use in the interpolation computation (i.e. the computation will be identical to if the computed output_size were passed-in explicitly). Note that when scale_factor is floating-point, the recomputed scale_factor may differ from the one passed in due to rounding and precision issues.

I can't understand what it means after reading many times.  Some confusions: 
+ > When scale_factor is passed as a parameter, it is used to compute the output_size. 
   
  What does `parameter` mean? Like a trainable parameter?  If this just says that "if `scale_factor` is given, output size will be computed using it", then shouldn't this sentence be in `scale_factor` doc instead?. 

+ >  If recompute_scale_factor is False or not specified, the passed-in scale_factor will be used in the interpolation computation.

   This says that if `recompute_scale_factor=True`, `scale_factor` will *not* be used? How does that make sense?

+ > Otherwise, a new scale_factor will be computed based on the output and input sizes for use in the interpolation computation (i.e. the computation will be identical to if the computed output_size were passed-in explicitly).

   The above sentence just said that `scale_factor` will not be used then, but how  would you get the output size if that is the case? Does it mean that I should use `size` instead? 

Additionally, this arg doc is also bad
> scale_factor (float or Tuple[float]) – multiplier for spatial size. Has to match input size if it is a tuple.

should be "If it is a tuple, its size has to match input dimensions".

cc @brianjo @mruberry @albanD @jbschlosser

### PR DESCRIPTION
**Pull request by thomasjpfan**
Fixes #55909

This PR looks to improve the documentation to describe the following behavior:

https://github.com/pytorch/pytorch/blob/8130f2f67ada1951ee27e55b8a506d6de23c13df/torch/nn/functional.py#L3673-L3685

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -3536,7 +3536,8 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
         input (Tensor): the input tensor
         size (int or Tuple[int] or Tuple[int, int] or Tuple[int, int, int]):
             output spatial size.
-        scale_factor (float or Tuple[float]): multiplier for spatial size. Has to match input size if it is a tuple.
+        scale_factor (float or Tuple[float]): multiplier for spatial size. If `scale_factor` is a tuple,
+            its length has to match `input`'s dimensions.
```
Comment (jbschlosser) [inline torch/nn/functional.py:3540]: minor nit: what do you think of "its length has to match `input.dim()`" instead?

File: torch/nn/functional.py [hunk 2]
```diff
@@ -3551,14 +3552,13 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
...
-            Otherwise, a new `scale_factor` will be computed based on the output and input sizes for
-            use in the interpolation computation (i.e. the computation will be identical to if the computed
-            `output_size` were passed-in explicitly).  Note that when `scale_factor` is floating-point,
-            the recomputed scale_factor may differ from the one passed in due to rounding and precision
-            issues.
+            interpolation calculation. If `recomputed_scale_factor` is ``True``, then
+            `scale_factor` must be passed in and `scale_factor` is used to compute the
+            output `size`. The computed output `size` will be used to infer new scales for
+            the interpolation. Note that when `scale_factor` is floating-point,
+            the recomputed scale_factor may differ from the one passed in due to rounding and
```
Comment (jbschlosser) [inline torch/nn/functional.py:3559]: formatting nit: backticks around scale_factor

File: torch/nn/functional.py [hunk 3]
```diff
@@ -3551,14 +3552,13 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
...
         recompute_scale_factor (bool, optional): recompute the scale_factor for use in the
-            interpolation calculation.  When `scale_factor` is passed as a parameter, it is used
-            to compute the `output_size`.  If `recompute_scale_factor` is ``False`` or not specified,
-            the passed-in `scale_factor` will be used in the interpolation computation.
-            Otherwise, a new `scale_factor` will be computed based on the output and input sizes for
-            use in the interpolation computation (i.e. the computation will be identical to if the computed
-            `output_size` were passed-in explicitly).  Note that when `scale_factor` is floating-point,
-            the recomputed scale_factor may differ from the one passed in due to rounding and precision
-            issues.
+            interpolation calculation. If `recomputed_scale_factor` is ``True``, then
```
Comment (jbschlosser) [inline torch/nn/functional.py:3555]: spelling nit: recomputed -> recompute throughout

File: torch/nn/functional.py [hunk 4]
```diff
@@ -3551,14 +3552,13 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
...
+            `scale_factor` must be passed in and `scale_factor` is used to compute the
+            output `size`. The computed output `size` will be used to infer new scales for
+            the interpolation. Note that when `scale_factor` is floating-point,
```
Comment (jbschlosser) [inline torch/nn/functional.py:3558]: minor rewording: what do you think of "Note that when `scale_factor` is floating-point, it may differ from the recomputed `scale_factor` due to rounding and precision issues." ?

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! Your version explains what's going on much better :) I have to admit it's a fairly confusing arg in general.

**Review by jbschlosser — APPROVED** [review]
LGTM!
